src/lib/utils/os_utils.cpp

```diff
@@ -614,6 +620,35 @@ void OS::page_named(void* page, size_t size) {
 #endif
 }
 
+#if defined(BOTAN_TARGET_OS_HAS_THREADS)
+void OS::set_thread_name(std::thread& thread, const char* const name) {
+   int r;
+   auto hdl = thread.native_handle();
```

Comment:
`std::thread::native_handle` isn't necessarily guaranteed to exist, which would cause this to fail to compile on a (probably weird) OS that supported threads but didn't let you access native handles.

Instead call `thread.native_handle()` within each block which ensures we only call it if we have some idea what OS we are on.

Reply:
ah right I thought HAS_THREADS meant "all the way", fair enough then.